packages/eas-cli/src/metadata/download.ts

```diff
@@ -54,8 +54,13 @@ export async function downloadMetadataAsync(metadataCtx: MetadataContext): Promi
     }
   }
 
-  await fs.writeJson(filePath, config.toSchema(), { spaces: 2 });
-  unsubscribeTelemetry();
+  try {
+    await fs.writeJson(filePath, config.toSchema(), { spaces: 2 });
+  } catch (error) {
+    throw error;
```

Comment:
this catch does not do anything, plus in my suggestion try block started just after subscribe

Reply:
Finally is still getting executed in this case (and thus unsubscribe), even when re-throwing the error. We still want to surface IO errors in case we run into them when writing the `store.config.json`.

But the async methods that might be throwing are all try-catched already. The tasks are permissive in failures, if one part fails we still attempt to do the other tasks. So another try-catch on everything after subscribing would only confuse other ppl I believe.

Another approach that might make more sense, and does what you suggested, is creating a `withTelemetryAsync(action: async () => Promise<void>)`. But not sure if I can get that in before the conf

Reply:
I'm not sure how what you are saying relates to my comment
- your current catch block does not do anything so you should just remove it, nothing will change in the behavior
- as for wrapping everything in try block, if you initialize some resource that has a cleanup try block should start immediately after that call. I assume most of the calls between subscribing and unsubscribing are unlikely to throw any error, but it actually improves readability because you immediately know that this cleanup will always happen.

`withTelemetryAsync` is also a good option, but I don't think it's necessary

Reply:
Ok, I'll put the `withTelemetryAsync` on my list to refactor to. For now, I removed the catch block and will merge it as-is.